From fcf65480ca7e18910268a17ce8ed5fa8c32830c5 Mon Sep 17 00:00:00 2001
From: Xiang Xu <xuxiang@eswincomputing.com>
Date: Mon, 7 Jul 2025 11:02:05 +0800
Subject: [PATCH 606/652] fix:cpufreq switch failed

Changelogs:
1. cpufreq  switch failed because not set voltage gpio
2. Correct spelling errors of words

Signed-off-by: Xiang Xu <xuxiang@eswincomputing.com>
---
 drivers/clk/eswin/clk.c | 60 ++++++++++++++++++++---------------------
 drivers/clk/eswin/clk.h |  2 +-
 2 files changed, 31 insertions(+), 31 deletions(-)

diff --git a/drivers/clk/eswin/clk.c b/drivers/clk/eswin/clk.c
index c4dc231f62d8b..f5284cb1b611d 100644
--- a/drivers/clk/eswin/clk.c
+++ b/drivers/clk/eswin/clk.c
@@ -140,13 +140,13 @@ int eswin_clk_register_fixed_rate(const struct eswin_fixed_rate_clock *clks,
 }
 EXPORT_SYMBOL_GPL(eswin_clk_register_fixed_rate);
 
-static int eswin_clk_set_cpu_volatge(struct gpio_desc *cpu_voltage_gpio,
-				     enum voltage_level target_volatge)
+static int eswin_clk_set_cpu_voltage(struct gpio_desc *cpu_voltage_gpio,
+				     enum voltage_level target_voltage)
 {
 	if (!cpu_voltage_gpio)
-		return -EINVAL;
+		return -ENODEV;
 
-	switch (target_volatge) {
+	switch (target_voltage) {
 	case VOLTAGE_0_9V:
 		gpiod_set_value(cpu_voltage_gpio, 1);
 		break;
@@ -154,8 +154,8 @@ static int eswin_clk_set_cpu_volatge(struct gpio_desc *cpu_voltage_gpio,
 		gpiod_set_value(cpu_voltage_gpio, 0);
 		break;
 	default:
-		pr_err("%s %d: unsupport  volatge %d\n", __func__, __LINE__,
-		       target_volatge);
+		pr_err("%s %d: unsupport  voltage %d\n", __func__, __LINE__,
+		       target_voltage);
 		return -EINVAL;
 	}
 	return 0;
@@ -319,7 +319,7 @@ static int clk_pll_set_rate(struct clk_hw *hw, unsigned long rate,
 	char clk_cpu_mux_name[50] = { 0 };
 	char clk_cpu_lp_pll_name[50] = { 0 };
 	char clk_cpu_pll_name[50] = { 0 };
-	enum voltage_level cpu_target_volatge = VOLTAGE_0_8V;
+	enum voltage_level cpu_target_voltage = VOLTAGE_0_8V;
 	bool force_1_8ghz = clk->force_1_8ghz;
 
 	ret = eswin_calc_pll(&frac_val, &postdiv1_val, &fbdiv_val, &refdiv_val,
@@ -395,18 +395,18 @@ static int clk_pll_set_rate(struct clk_hw *hw, unsigned long rate,
 		switch (rate) {
 		case CLK_FREQ_1800M:
 		case CLK_FREQ_1700M:
-			cpu_target_volatge = VOLTAGE_0_9V;
+			cpu_target_voltage = VOLTAGE_0_9V;
 			break;
 		case CLK_FREQ_1600M:
 		case CLK_FREQ_1500M:
-			cpu_target_volatge = true == g_cpu_info.cpu_no_boost_1_6ghz ?
+			cpu_target_voltage = true == g_cpu_info.cpu_no_boost_1_6ghz ?
 						     VOLTAGE_0_8V :
 						     VOLTAGE_0_9V;
 			if (force_1_8ghz)
-				cpu_target_volatge = VOLTAGE_0_9V;
+				cpu_target_voltage = VOLTAGE_0_9V;
 			break;
 		default:
-			cpu_target_volatge = VOLTAGE_0_8V;
+			cpu_target_voltage = VOLTAGE_0_8V;
 			/*
 			 * For boards that do not support voltage switching,
 			 * the voltage is maintained at 0.8V.
@@ -416,23 +416,23 @@ static int clk_pll_set_rate(struct clk_hw *hw, unsigned long rate,
 		}
 	}
 	mutex_unlock(&lock);
-	
-	ret = eswin_clk_set_cpu_volatge(clk->cpu_voltage_gpio,
-					cpu_target_volatge);
-	if (ret && !force_1_8ghz) {
-		pr_warn("failed to change cpu volatge to %d mV, not support rate %ld\n",
-			cpu_target_volatge, rate);
-		goto switch_back;
-	} else {
-		if (clk->cpu_current_volatge !=
-			cpu_target_volatge) {
-			pr_info("cpu volatge change to %d mV, target rate %ld\n",
-				cpu_target_volatge, rate);
-			clk->cpu_current_volatge =
-				cpu_target_volatge;
+
+	if (clk->cpu_current_voltage !=	cpu_target_voltage) {
+		ret = eswin_clk_set_cpu_voltage(clk->cpu_voltage_gpio,
+						cpu_target_voltage);
+		if (ret && !force_1_8ghz) {
+			pr_warn("failed to change cpu voltage to %d mV, not support rate %ld\n",
+				cpu_target_voltage, rate);
+			goto switch_back;
+		} else {
+			pr_info("cpu voltage change to %d mV, target rate %ld\n",
+				cpu_target_voltage, rate);
+			clk->cpu_current_voltage =
+				cpu_target_voltage;
 		}
 	}
 
+
 	/*first disable pll */
 	val = readl_relaxed(clk->ctrl_reg0);
 	val &= ~(((1 << clk->pllen_width) - 1) << clk->pllen_shift);
@@ -672,11 +672,11 @@ void eswin_clk_register_pll(struct eswin_pll_clock *clks, int nums,
 			devm_gpiod_get(dev, "cpu-voltage", GPIOD_OUT_HIGH) :
 			cpu_voltage_gpio;
 	if (IS_ERR_OR_NULL(cpu_voltage_gpio)) {
-		dev_warn(dev, "failed to get cpu volatge gpio\n");
+		dev_warn(dev, "failed to get cpu voltage gpio, unable to adjust CPU voltage\n");
 		cpu_voltage_gpio = NULL;
 	} else {
-		/*cpu default freq is 1400M, the volatge should be VOLTAGE_0_8V*/
-		eswin_clk_set_cpu_volatge(cpu_voltage_gpio, VOLTAGE_0_8V);
+		/*cpu default freq is 1400M, the voltage should be VOLTAGE_0_8V*/
+		eswin_clk_set_cpu_voltage(cpu_voltage_gpio, VOLTAGE_0_8V);
 	}
 
 	force_1_8ghz = device_property_read_bool(dev, "force-1_8ghz");
@@ -746,8 +746,8 @@ void eswin_clk_register_pll(struct eswin_pll_clock *clks, int nums,
 
 		p_clk->hw.init = &init;
 		p_clk->cpu_voltage_gpio = cpu_voltage_gpio;
-		p_clk->cpu_current_volatge = VOLTAGE_0_8V;
-                p_clk->force_1_8ghz = force_1_8ghz;
+		p_clk->cpu_current_voltage = VOLTAGE_0_8V;
+		p_clk->force_1_8ghz = force_1_8ghz;
 		clk = clk_register(dev, &p_clk->hw);
 		if (IS_ERR(clk)) {
 			devm_kfree(dev, p_clk);
diff --git a/drivers/clk/eswin/clk.h b/drivers/clk/eswin/clk.h
index 167e2a627a7a1..42cf07d693ec1 100644
--- a/drivers/clk/eswin/clk.h
+++ b/drivers/clk/eswin/clk.h
@@ -162,7 +162,7 @@ struct eswin_clk_pll {
 	u8 lock_shift;
 	u8 lock_width;
 	struct gpio_desc *cpu_voltage_gpio;
-	enum voltage_level cpu_current_volatge;
+	enum voltage_level cpu_current_voltage;
 	bool force_1_8ghz;
 };
 
-- 
2.51.0

